taskcluster/async/asyncclient.py

```diff
@@ -152,6 +147,7 @@ def __del__(self):
             status = response.status
             if status == 204:
                 return None
+            rerr = await response.text()
```

Comment:
Actually, let's match the sync class and do  `log.warn('Retrying because of a %s status code' % status)`